### PR TITLE
Fixes numerical accuracy issues in quantile.

### PR DESCRIPTION
**Pull request by simonbyrne**
Fixes issue https://github.com/JuliaStats/StatsBase.jl/issues/164, and another when `p < eps()`.

### CODE REVIEW

**Comment by simonbyrne** [comment]
cc @andreasnoack can you review and merge?

File: base/statistics.jl [hunk 1]
```diff
@@ -577,19 +577,21 @@ end
...
     lv = length(v)
-    index = 1 + (lv-1)*p
-    1 <= index <= lv || error("input probability out of [0,1] range")
+    f0 = (lv-1)*p # 0-based interpolated index
+    t0 = trunc(f0)
+    h = f0 - t0
 
-    indlo = floor(index)
-    i = trunc(Int,indlo)
+    i = trunc(Int,t0) + 1
```
Comment (andreasnoack) [inline base/statistics.jl]: To avoid confusion, maybe just `t0 = Int(f0)` since `f0` has already been truncated.

**Comment by andreasnoack** [comment]
Except for the minor comment, it looks good.

File: base/statistics.jl (hunk 1 above)
Comment (simonbyrne) [inline base/statistics.jl]: `trunc(Int, t0)` is technically faster, as it avoids the check of an integer.

Comment (andreasnoack) [inline base/statistics.jl]: Makes sense.

**Pull request merged**